CDH-3581: Initial version of release notes (#172)
SoftwareAG/c8y-releasenotes: content/release-10-17-0/announcements-10-17-0.md

`sag_apama_correlator_user_httpServer_instance_5_metricname`

### Cumulocity IoT DataHub

#### Planned

##### Breaking change in the offloading mechanism - deprecation of support for mixed types

> In a future release of Cumulocity IoT DataHub a newer version of Dremio will be adopted that removes support for mixed data types in a single column. The offloading process will be adapted so that mixed types will be automatically resolved during offloading. Cumulocity IoT DataHub will not be able to read data lake contents containing mixed data types, so corresponding data needs to be converted.

Author: Done

##### Breaking change in the offloading mechanism - switch to new table format may introduce incompatibilities

In future releases of Cumulocity IoT DataHub the offloading mechanism may leverage the Apache Iceberg table format, which is an open format supported by Dremio. Due to this change, current offloading configurations may fail. For example, they may fail due to rarely used data types which are incompatible with the Apache Iceberg format. Further details and remediation actions will be provided closer to the respective release.

##### Breaking change in the offloading mechanism - removal of history column in alarms collection

> In a future release of Cumulocity IoT core the history column will be removed from the alarms collection. Therefore, Cumulocity IoT DataHub does no longer include this column in newly defined offloadings. If alarms data has been already offloaded into the data lake, the Dremio table associated with that data lake folder still includes the history column as part of its schema. With new offloading runs the values of this column will be null.

Author: Done

> In a future release of Cumulocity IoT core the history column will be removed from the alarms collection. Therefore, Cumulocity IoT DataHub will not include this column in newly defined offloadings anymore. If alarms data has been already offloaded into the data lake, the Dremio table associated with that data lake folder will still include the history column as part of its schema. With new offloading runs the values of this column will be null.

Reviewer: ```suggestion
In a future release of Cumulocity IoT core the history column will be removed from the alarms collection. Therefore, Cumulocity IoT DataHub does no longer include this column in newly defined offloadings. If alarms data has been already offloaded into the data lake, the Dremio table associated with that data lake folder still includes the history column as part of its schema. With new offloading runs the values of this column will be null.
```

Author: Done
